Sources/OpenAPIRuntime/URICoder/Serialization/URISerializer.swift

```diff
@@ -228,8 +232,15 @@ extension URISerializer {
         case (.simple, false):
             keyAndValueSeparator = ","
             pairSeparator = ","
+        case (.deepObject, _):
+            keyAndValueSeparator = "="
+            pairSeparator = "&"
         }
 
+        func serializeNestedKey(_ elementKey: String, forKey rootKey: String) -> String {
+            guard case .deepObject = configuration.style else { return elementKey }
+            return rootKey + "[" + elementKey + "]"
```

Comment:
Can you check that we end up percent-escaping the `[` and `]` characters? Technically they should be escaped.

However, when parsing, we should be flexible and support both escaped and unescaped.

Reply:
Thanks for adding the test verifying this.